hb_report/utillib.py

```diff
@@ -1649,11 +1649,29 @@ def time_status():
     time_f = os.path.join(constants.WORKDIR, constants.TIME_F)
     crmutils.str2file(out_string, time_f)
 
+
+def dump_D_process():
+    '''
+    dump D-state process stack
+    '''
+    out_string = ""
+    _, out, _ = crmutils.get_stdout_stderr("ps aux|awk '$8 ~ /^D/{print $2}'")
+    len_D_process = len(out.split('\n')) if out else 0
+    out_string += "Dump D-state process stack: {}\n".format(len_D_process)
+    if len_D_process == 0:
+        return out_string
+    for pid in out.split('\n'):
+        _, cmd_out, _ = crmutils.get_stdout_stderr("cat /proc/{}/comm".format(pid))
+        out_string += "pid: {}     comm: {}\n".format(pid, cmd_out)
```

Comment:
I would implement everything with the `format` method without the `+=`, but this is OK anyway.
(`out_string = "{}pid: {}     comm: {}\n".format(out_string, pid, cmd_out)